rustpad-server/src/database.rs

```diff
@@ -49,10 +49,13 @@ impl Database {
     pub async fn store(&self, document_id: &str, document: &PersistedDocument) -> Result<()> {
         let result = sqlx::query(
             r#"
-INSERT OR REPLACE INTO
+INSERT INTO
     document (id, text, language)
 VALUES
-    ($1, $2, $3)"#,
+    ($1, $2, $3)
+ON CONFLICT(id) DO UPDATE SET
+    text = excluded.text,
+    language = excluded.language"#,
```

Comment:
Is this necessary or just an optimization for rowid? If just a minor optimization then I would prefer not changing.

Reply:
I'm running rustpad on [mvsqlite](https://github.com/losfair/mvsqlite) and although not necessary this is a important optimization. `rowid` allocation is a high-contention operation - two transactions that both increments max `rowid` can never be concurrently executed.

It's also the case with SQLite's own [Begin Concurrent](https://www.sqlite.org/cgi/src/doc/begin-concurrent/doc/begin_concurrent.md) branch.

Reply:
Got it, thank you for teaching me about this! :)